src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,273 @@
+     * directly impact customer-facing features. Applications may use
+     * loss of Production Critical events to indicate system failure.
+     * On constrained devices, entries logged with Critical
+     * priority must be accounted for in the power and memory budget,
+     * as it is expected that they are always logged and offloaded
+     * from the device.
+     */
+    Critical = kPriorityLevel_First,
+    /**
+     * Info priority denotes log entries that provide extra insight
+     * and diagnostics into the running system. Info logging level may
+     * be used over an extended period of time in a production system,
+     * or may be used as the default log level in a field trial. On
+     * the constrained devices, the entries logged with Info level must
+     * be accounted for in the bandwidth and memory budget, but not in
+     * the power budget.
+     */
+    Info = 2,
+
+    /**
+     *  Debug priority denotes log entries of interest to the
+     *  developers of the system and is used primarily in the
+     *  development phase. Debug priority logs are
+     *  not accounted for in the bandwidth or power budgets of the
+     *  constrained devices; as a result, they must be used only over
+     *  a limited time span in production systems.
+     */
+    Debug                  = 3,
+    kPriorityLevel_Last    = Debug,
+    kPriorityLevel_Invalid = 4,
+
+} PriorityLevel;
+
+/**
+ * @brief
+ *   The structure that defines a schema for event metadata.
+ */
+struct EventSchema
+{
+    chip::ClusterId mClusterId; ///< ID of cluster
+    uint32_t mStructureType;    ///< Type of structure
+    PriorityLevel mPriority;    ///< Priority
+};
+
+/**
+ * @typedef timestamp_t
+ * Type used to describe the timestamp in milliseconds.
+ */
+typedef uint32_t timestamp_t;
+
+/**
+ * @typedef duration_t
+ * Type used to describe the duration, in milliseconds.
+ */
+typedef uint32_t duration_t;
+
+/**
+ * @typedef utc_timestamp_t
+ * Type used to describe the UTC timestamp in milliseconds.
+ */
+typedef uint64_t utc_timestamp_t;
+
+enum
+{
+    kTag_Region  = 1, ///< An 32-bit unsigned indicating the log region, i.e. the module to which the log message pertains.
+    kTag_Message = 2  ///< A string containing the actual debug message.
+};
+
+// Structures used to describe in detail additional options for event encoding
+
+/**
+ * @brief
+ *   The structure that provides a full resolution of the cluster.
+ */
+struct DetailedRootSection
+{
+    /**
+     * Default constructor
+     */
+    DetailedRootSection(void){};
+    chip::NodeId mNodeId;
+    chip::EndpointId mEndpointId;
+};
+
+/**
+ * @brief
+ *   The validity and type of timestamp included in EventOptions.
+ */
+typedef enum TimestampType
+{
+    kTimestampType_Invalid = 0,
+    kTimestampType_System,
+    kTimestampType_UTC
+} TimestampType;
+
+/**
+ * @brief
+ *   The union that provides an application set system or UTC timestamp.
+ */
+union Timestamp
```

Comment:
how does one know which timestamp is to be used?

This is a union of two identical values, but their logic is different.

Reply:
if you see EventOptions, there is TimestampType timestampType, when application call LogEvent with EventOptions , which would log events to circular buffer, 
it will do below logic(I have nott yet im)
    if ((apOptions != NULL) && (apOptions->timestampType == kTimestampType_System))
    {
        opts.timestamp.systemTimestamp = apOptions->timestamp.systemTimestamp;
    }
(I am not implementing utc support in this PR, and it would next PRs)
    if ((inOptions != NULL) && (inOptions->timestampType == kTimestampType_UTC))
    {
        opts.timestamp.utcTimestamp = inOptions->timestamp.utcTimestamp;
        opts.timestampType          = kTimestampType_UTC;
    }

Reply:
Just follow your suggestion for improvement, thanks